src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadYamlStep.java

```diff
@@ -116,6 +131,15 @@ public static int getDefaultMaxAliasesForCollections() {
 		return DEFAULT_MAX_ALIASES_FOR_COLLECTIONS;
 	}
 
+	public int getCodePointLimit() {
+		return codePointLimit;
+	}
+
+	@DataBoundSetter
+	public void setCodePointLimit(final int codePointLimit) {
+		this.codePointLimit = codePointLimit;
```

Comment:
This is very similar to `masAliasForCollection`. So you need to add similar protections as in `setMaxAliasesForCollections(...)` or a potential nefarious user could sink a Jenkins controller by reading in a couple of GB of data until the instance runs our of memory.

Reply:
Whoever sets this limit can also just execute random groovy code and sink the instance this way.

Reply:
No because the pipeline is running in the sandbox.
https://www.jenkins.io/doc/developer/security/misc/#groovy-scripting

Reply:
Besides, it is needed for users to not shoot themselves in the foot. They don't know how much memory the Jenkins controller has available, but the administrator should.